### PR TITLE
HW5.1 solution

### PR DESCRIPTION
**Pull request by vladspirin**
It was an exciting task. I found a filter() and used it here, but I spent a lot of time mixing my different solutions. 
I think there is a better way to solve this task. Using function, for example.

### CODE REVIEW

**Comment by vladspirin** [comment]
I changed my code to the new one without a filter(). But now, two linter issues returned 'W504 line break after binary operator.' 
I don't understand why Linter skipped the PEP8 standard 
Found this https://peps.python.org/pep-0008/#indentation
I disagree in this case with this Linter

**Review by antohaUa — APPROVED** [review]

File: lesson_05/homework5_1.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            k: v for k, v in dict_copy.items()
+            if (
+                v[2] >= search_criteria[1] and
+                v[1] >= search_criteria[0] and
+                v[4] <= search_criteria[2]
+            )
+        }.items(),
+        key=lambda x: x[1][4])[:5],
+)
```
[the remark this answers is not in the document]
Comment (vladspirin) [reply lesson_05/homework5_1.py:70]: I'll rewrite this solution soon

**Comment by vladspirin** [comment]
I refactored my code. As you said, I used functions and separated my code. Now, it looks scalable and has better readability. However, I think there is still a lot of room to improve my code.

File: lesson_05/homework5_1.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+_log = logging.getLogger(__name__)
+# Exists some car data with color, year, engine_volume, car type , price
+# We have search_criteria as tuple of ( year>= , engine_volume >= , price<=)
+# write code that will help us to get cars that satisfy search_criteria.
+# Cars should be sorted by price ascending.
+# We should print up to five (5) first found elements
+
+
+# 1. condition to check the criteria
+def check_car_criteria(car, year, engine_size, max_price):
```
Comment (antohaUa) [inline lesson_05/homework5_1.py:16]: The best option to pass here jusr car and search_criteria

File: lesson_05/homework5_1.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    Args:
+        car (str): string.
+        year (int): integer.
+        engine_size (float): float.
+        max_price (int): integer.
+    Returns:
+        bool: Result of condition.
+    """
+    color, car_year, engine, car_type, price = car
```
Comment (antohaUa) [inline lesson_05/homework5_1.py:28]: For not used variables do not forget to use not names but underscope "_"

**Review by antohaUa — APPROVED** [review]

**Pull request merged**